🐛  [BUG] Code generator doesn't mark path parameters as required, causing TypeScript errors

## Bug Description

The OpenAPI code generator produces TypeScript hooks where path parameters are treated as optional, causing `tsc` errors under `strict: true` mode. The generated `useX` and `useXSuspense` hooks pass `options?.params` (which is `ParamType | undefined`) to fetch functions that require `ParamType` (non-undefined).

## Reproduction

1. Create a FastAPI route with path parameters:
```python
@router.get("/scans/{scan_id}", response_model=ScanOut, operation_id="getScan")
def get_scan(scan_id: str, store: Dependency.Store):
    ...
```

2. Run `apx dev check` — the generated `lib/api.ts` will produce errors like:
```
src/.../ui/lib/api.ts(312,83): error TS2345: Argument of type 'GetScanParams | undefined' 
is not assignable to parameter of type 'GetScanParams'.
  Type 'undefined' is not assignable to type 'GetScanParams'.
```

## Generated Code (Problematic)

```typescript
// The fetch function requires non-optional params:
export const getScan = async (params: GetScanParams, options?: RequestInit) => { ... };

// But the hook passes options?.params which can be undefined:
export function useGetScan<TData = ...>(options?: { params?: GetScanParams; ... }) {
  return useQuery({ 
    queryKey: getScanKey(options?.params), 
    queryFn: () => getScan(options?.params),  // ❌ TS2345
    ...options?.query 
  });
}
```

## Expected Behavior

Per the OpenAPI 3.0 specification, **path parameters are always required** ([spec ref](https://spec.openapis.org/oas/v3.0.3#parameter-object)). The generator should either:

1. Force `required: true` for path parameters in `normalize_param()` regardless of what the OpenAPI schema says (since `#[serde(default)]` on the `required` field defaults to `false` when not explicitly set), **or**
2. Generate hooks with non-null assertions: `options!.params!`, **or**
3. Make `params` required (not optional) in hooks when the underlying fetch function requires them

## Root Cause

In `crates/core/src/openapi/ir/normalize.rs`, the `normalize_param` function trusts the `required` field from the OpenAPI spec:

```rust
ParamIR {
    // ...
    required: p.required,  // defaults to false via #[serde(default)]
    // ...
}
```

And in `crates/core/src/openapi/spec.rs`:

```rust
#[serde(default)]  // defaults to false
pub required: bool,
```

The fix should enforce `required: true` for path parameters:

```rust
let required = if location == ParamLocation::Path {
    true  // OpenAPI 3.0 mandates path params are always required
} else {
    p.required
};
```

## Workaround

Running `npx tsc --noEmit` after manually patching the generated file with `options!.params!` works, but `apx dev check` regenerates the file before running tsc, making manual patches ineffective.

## Environment

- apx version: 0.2.1
- TypeScript: 5.9.3 (strict mode)
- FastAPI backend with path parameters

--- a/crates/core/src/openapi/ir/normalize.rs
+++ b/crates/core/src/openapi/ir/normalize.rs
@@ -617,11 +617,18 @@ fn normalize_param(p: &Parameter) -> ParamIR {
         _ => ParamLocation::Query,
     };
 
+    // OpenAPI 3.0 spec: path parameters are always required
+    let required = if location == ParamLocation::Path {
+        true
+    } else {
+        p.required
+    };
+
     ParamIR {
         name: p.name.clone(),
         original_name: p.name.clone(),
         ty,
-        required: p.required,
+        required,
         location,
     }
 }

--- a/crates/core/src/openapi/mod.rs
+++ b/crates/core/src/openapi/mod.rs
@@ -4011,4 +4011,51 @@ export function useGetCatBuggy<TData = { data: { meow: string } }>(
         // Should generate functions without any types section
         assert!(ts_code.contains("ping"), "Should generate ping function");
     }
+
+    #[test]
+    fn test_path_params_required_even_when_omitted() {
+        // Path parameters omit `required` (common in FastAPI-generated specs).
+        // Per OpenAPI 3.0 spec, path params are always required.
+        let openapi_json = r##"{
+  "openapi": "3.1.0",
+  "info": { "title": "Test API", "version": "1.0.0" },
+  "paths": {
+    "/scans/{scan_id}": {
+      "get": {
+        "operationId": "getScan",
+        "parameters": [
+          { "name": "scan_id", "in": "path", "schema": { "type": "string" } }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": { "application/json": { "schema": { "type": "object", "properties": { "id": { "type": "string" } } } } }
+          }
+        }
+      }
+    }
+  }
+}"##;
+
+        let ts_code = generate_and_verify(openapi_json);
+        println!("=== PATH PARAMS REQUIRED ===\n{ts_code}\n=== END ===");
+
+        // Fetch function: params must be non-optional
+        assert!(
+            ts_code.contains("params: GetScanParams, options?"),
+            "Fetch function params should be required (non-optional)"
+        );
+
+        // Hook: options object must be required (contains required params)
+        assert!(
+            ts_code.contains("(options: { params: GetScanParams"),
+            "Hook options should be required with required params field"
+        );
+
+        // Hook should use options.params, not options?.params
+        assert!(
+            ts_code.contains("options.params"),
+            "Should use options.params (non-optional access)"
+        );
+    }
 }

